bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/handler/MeteoAlerteHandler.java

```diff
@@ -0,0 +1,215 @@
+                    jsonDeserializationContext) -> ZonedDateTime.parse(json.getAsJsonPrimitive().getAsString()))
+            .create();
+
+    // Time zone provider representing time zone configured in openHAB configuration
+    private final TimeZoneProvider timeZoneProvider;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private String queryUrl = "";
+
+    public MeteoAlerteHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Météo Alerte handler.");
+
+        MeteoAlerteConfiguration config = getConfigAs(MeteoAlerteConfiguration.class);
+        logger.debug("config department = {}", config.department);
+        logger.debug("config refresh = {}", config.refresh);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        queryUrl = URL + config.department;
+        refreshJob = scheduler.scheduleWithFixedDelay(this::updateAndPublish, 0, config.refresh, TimeUnit.HOURS);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Météo Alerte handler.");
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateAndPublish();
+        } else {
+            logger.debug("The Meteo Alerte binding is read-only and can not handle command {}", command);
+        }
+    }
+
+    private void updateAndPublish() {
+        try {
+            if (queryUrl.isEmpty()) {
+                throw new MalformedURLException("queryUrl not initialized");
+            }
+            String response = HttpUtil.executeUrl("GET", queryUrl, TIMEOUT_MS);
+            updateStatus(ThingStatus.ONLINE);
+            ApiResponse apiResponse = gson.fromJson(response, ApiResponse.class);
+            updateChannels(apiResponse);
+        } catch (MalformedURLException e) {
+            logger.warn("Malformed URL in Météo Alerte request : {}", queryUrl);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        } catch (IOException e) {
+            logger.warn("Error opening connection to Meteo Alerte webservice : {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     * Update the channel from the last Meteo Alerte data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    private void updateChannels(ApiResponse apiResponse) {
+        Arrays.stream(apiResponse.getRecords()).findFirst().ifPresent(record -> {
```

Comment:
```suggestion
        Arrays.stream(apiResponse.getRecords()).findFirst().ifPresent(record ->
```

statement lambda can be replaced by expression lambda if only a single expression ins present.